fs/base.py

```diff
@@ -1161,6 +1175,9 @@ def move(self, src_path, dst_path, overwrite=False, preserve_time=False):
             raise errors.DestinationExists(dst_path)
         if self.getinfo(src_path).is_dir:
             raise errors.FileExpected(src_path)
+        if normpath(src_path) == normpath(dst_path):
+            # early exit when moving a file onto itself
+            return
```

Comment:
Why use `normpath` here when you started with a `self.validatepath` call for `src_path` and `dst_path` in the other functions?